### PR TITLE
fix: Selected Items of Ticket Fragment do not retain on screen rotation

### PR DESCRIPTION
**Pull request by pranavpandey1998official**
Lifted ticketIdAndQty from view layer to viewModel and done required changes

feat: #1097
fixes: #965 
fixes: #1191

### CODE REVIEW

**Comment by pranavpandey1998official** [comment]
done

File: app/src/main/java/org/fossasia/openevent/general/ticket/TicketsFragment.kt [hunk 1]
```diff
@@ -122,9 +123,15 @@ class TicketsFragment : Fragment() {
...
+        ticketsViewModel.ticketIdAndQty
+            .nonNull()
+            .observe(this, Observer {
+                ticketsRecyclerAdapter.addAllTicketIdAndQty(it)
+                ticketsRecyclerAdapter.notifyDataSetChanged()
+            })
 
         rootView.register.setOnClickListener {
-            if (!ticketsViewModel.totalTicketsEmpty(ticketIdAndQty)) {
+            if (!ticketsViewModel.totalTicketsEmpty(ticketsViewModel.ticketIdAndQty.value!!)) {
```
Comment (liveHarshit) [inline app/src/main/java/org/fossasia/openevent/general/ticket/TicketsFragment.kt]: Do not use non null assertion.

File: app/src/main/java/org/fossasia/openevent/general/ticket/TicketsFragment.kt [hunk 2]
```diff
@@ -146,7 +153,7 @@ class TicketsFragment : Fragment() {
     private fun redirectToAttendee() {
         val bundle = Bundle()
         bundle.putLong(EVENT_ID, id)
-        bundle.putSerializable(TICKET_ID_AND_QTY, ticketIdAndQty)
+        bundle.putSerializable(TICKET_ID_AND_QTY, ticketsViewModel.ticketIdAndQty.value as Serializable)
```
[the remark this answers is not in the document]
Comment (liveHarshit) [reply app/src/main/java/org/fossasia/openevent/general/ticket/TicketsFragment.kt]: IMO, you can use null check. Cast is not needed.

**Comment by pranavpandey1998official** [comment]
@nikit19 @iamareebjamal, please review

**Comment by liveHarshit** [comment]
@pranavpandey1998official Any updates?

**Pull request closed without merging**